src/cr/cube/cube.py

```diff
@@ -422,38 +422,14 @@ def _ca_as_0th(self):
     def _cube_dict(self):
         """dict containing raw cube response, parsed from JSON payload."""
         cube_dict = copy.deepcopy(self._cube_response)
-        # ---cube inflation---
-        # ---In case of numeric arrays, we need to inflate the rows dimension
-        # ---according to the mean subvariables. For each subvar the rows dimension
-        # ---will have a new element related to the subvar metadata.
         if self._mean_subvariables:
             dimensions = cube_dict.get("result", {}).get("dimensions", {})
-            rows_dimension = {
-                "references": {"alias": "mean", "name": "mean"},
-                "type": {
-                    "elements": [],
-                    "class": "enum",
-                    "subtype": {"class": "variable"},
-                },
-            }
-            subrefs = self._mean_subreferences
-            for i, _ in enumerate(self._mean_subvariables):
-                rows_dimension["type"].get("elements", []).append(
-                    {
-                        "id": i,
-                        "value": {
-                            "references": {
-                                "alias": subrefs[i].get("alias") if subrefs else None,
-                                "name": subrefs[i].get("name") if subrefs else None,
-                            }
-                        },
-                    },
-                )
-            dimensions.append(rows_dimension)
+            dimensions.append(self._numeric_array_dimensions)
```

Comment:
Since we only have a single append here, I'd prefer a more functional expression here (and not mutation). Something like:
```
dimensions = cube_dict.get(. . .) + [self._numeric_array_dimension]
```
(provided that you change the name of that property).

Reply:
actaully we can't do this cause otherwise we don't mutate the cube_dict object